### PR TITLE
TINY-6656: Fixed inserting a table via the mceInsertTable command incorrectly creating 2 undo levels

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6656

Description of Changes:
This fixes calling the `mceInsertTable` command incorrectly creating 2 undo levels. This was caused by calling `editor.insertContent()` which creates a undo level and then further manipulating the dom after to ensure the table sizing is correct. This wasn't an issue with the menu item, as it was wrapped in an undo transaction (I've also made the menu item use the command to keep things consistent). This also fixes another issue I noticed where the plugin API wasn't creating an undo level.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by JamesToohey — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts [hunk 1]
```diff
@@ -13,69 +13,82 @@ UnitTest.asynctest('browser.tinymce.plugins.table.command.InsertTableCommandTest
   TinyLoader.setup((editor, onSuccess, onFailure) => {
     const tinyApis = TinyApis(editor);
 
+    const sResetContent = Step.sync(() => editor.resetContent(''));
+    const sAssertNumNewUndoLevels = (expected: number) => Step.sync(() => {
```
Comment (JamesToohey) [inline modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts:17]: I will note I found this naming a bit confusing because it sort of implies some kind of state (like if I call it once, then change something, and call it again, I'd get the difference rather than the length of all undo levels except the first)

Comment (lnewson) [reply modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts:17]: Open to suggestions for a better name? The idea here was to check the number of new undo levels from the initial state (which will always have 1 level).

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts [hunk 2]
```diff
@@ -43,12 +43,17 @@ const insert = (editor: Editor, columns: number, rows: number, colHeaders: numbe
...
+  // Don't create an undo level when inserting the base table HTML otherwise we can end up with 2 undo levels
+  editor.undoManager.ignore(() => {
+    const table = TableRender.render(rows, columns, rowHeaders, colHeaders, getTableHeaderType(editor), options);
+    Attribute.set(table, 'data-mce-id', '__mce');
 
-  const html = Html.getOuter(table);
-  editor.insertContent(html);
+    const html = Html.getOuter(table);
+    editor.insertContent(html);
+    editor.addVisual();
```
Comment (ashfordneil) [inline modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts:53]: Why was this added?

**Review by ashfordneil — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts (hunk 1 above)
Comment (ashfordneil) [reply modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts:17]: Some ideas:
1. We could just make it `sAssertNumUndoLevels` and add 1 to each number we assert it with
2. Could we tie a TestStore to undo events (if we have those) that way we can call `.sClear` and `.sAssert` as needed

File: modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts:53]: Because it's needed and was previously being run for the menu item?

File: modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts:17]: > We could just make it sAssertNumUndoLevels and add 1 to each number we assert it with
I originally did this, but it was just confusing to me.

> Could we tie a TestStore to undo events (if we have those) that way we can call .sClear and .sAssert as needed
We don't have this. @JamesToohey created a JIRA for us to look into adding this the other day.

File: modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts:53]: I should clarify, this adds the visual helpers (dotted border when a table has no borders), etc...

File: modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts (hunk 1 above)
Comment (metricjs) [reply modules/tinymce/src/plugins/table/test/ts/browser/command/InsertTableCommandTest.ts:17]: Only thing I can think of is `sAssertNumTestUndoLevels` and I'm not sure that's any better.

**Review by metricjs — APPROVED** [review]
